src/xcode/ENA/ENA/Source/AppDelegate.swift

```diff
@@ -93,7 +93,15 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 	let taskScheduler: ENATaskScheduler = ENATaskScheduler.shared
 
 	lazy var appConfigurationProvider: AppConfigurationProviding = {
-		CachedAppConfiguration(client: self.client)
+		// use a custom http client that uses/recognized caching mechanisms
+		let appFetchingClient = CachingHTTPClient(basedOn: client)
```

Comment:
To minimize the dependency would it be possible to inject the Configuration?

Reply:
It's not a hard dependency (on `client`). The caching client could also be initialized like `CachingHTTPClient()` and works the same. I used the existing 'scheme' to keep the changes simple. If you have something specific in mind, please suggest :)

(The AppDelegate definitely needs some love but that should be a dedicated refactoring/cleanup)

Reply:
The CachingHTTPClient only uses the HTTPClient.Configuration.
So I would suggest to only inject HTTPClient.Configuration.

Reply:
Ah, thanks for clarification. 

My initial idea was to use the original session as template as well (plus caching) but this changed. Will change this in a moment.